src/Sylius/Bundle/CoreBundle/Form/Type/ChannelPricing/ChannelPricingsType.php

```diff
@@ -0,0 +1,156 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Bundle\CoreBundle\Form\Type\ChannelPricing;
+
+use Sylius\Bundle\CoreBundle\Form\Type\Product\ChannelPricingType;
+use Sylius\Component\Channel\Repository\ChannelRepositoryInterface;
+use Sylius\Component\Core\Model\ChannelInterface;
+use Sylius\Component\Core\Model\ChannelPricing;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\Form\AbstractType;
+use Symfony\Component\Form\Extension\Core\Type\CollectionType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\Form\FormEvent;
+use Symfony\Component\Form\FormEvents;
+use Symfony\Component\Form\FormInterface;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+/**
+ * @author Mateusz Zalewski <mateusz.zalewski@lakion.com>
+ */
+final class ChannelPricingsType extends AbstractType implements EventSubscriberInterface
+{
+    /**
+     * @var ChannelRepositoryInterface
+     */
+    private $channelRepository;
+
+    /**
+     * @param ChannelRepositoryInterface $channelRepository
+     */
+    public function __construct(ChannelRepositoryInterface $channelRepository)
+    {
+        $this->channelRepository = $channelRepository;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function buildForm(FormBuilderInterface $builder, array $options)
+    {
+        $builder->addEventSubscriber($this);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public static function getSubscribedEvents()
+    {
+        return [
+            FormEvents::PRE_SET_DATA => 'preSetData',
+            FormEvents::SUBMIT => 'submit',
+        ];
+    }
+
+    /**
+     * @param FormEvent $event
+     */
+    public function preSetData(FormEvent $event)
+    {
+        $form = $event->getForm();
+        $allChannels = $this->channelRepository->findAll();
+
+        /** @var FormInterface $children */
+        foreach ($this->getChannelCodesToDelete($form, $allChannels) as $channelCode) {
+            $form->remove($channelCode);
+        }
+
+        /** @var ChannelInterface $channel */
+        foreach ($allChannels as $channel) {
+            if ($form->has($channel->getCode())) {
+                continue;
+            }
+
+            $form->add($channel->getCode(), ChannelPricingType::class, [
```

Comment:
Can we do not make dynamic fields like that? It makes customising form types much harder (/cc @NeverResponse), customising rendering of that field is impossible in this case.